### PR TITLE
Fix: Installing a package with versioned data 

### PR DESCRIPTION
**Pull request by mawtex**
Installing a package with versioned data on a site where this data already exists (in the same or a different version) would fail. This fix ensures VersionId is taken into account during the

### CODE REVIEW

File: Composite/Core/PackageSystem/PackageFragmentInstallers/DataPackageFragmentInstaller.cs [hunk 1]
```diff
@@ -281,12 +284,15 @@ private void UpdateVersionId(IVersioned data)
         }
 
 
-        private static DataKeyPropertyCollection CopyFieldValues(DataType dataType, IData data, XElement addElement)
+        private static DataPropertyValueCollection GetKeyPropertyValues(DataType dataType, IData data, XElement addElement)
```
Comment (napernik) [inline Composite/Core/PackageSystem/PackageFragmentInstallers/DataPackageFragmentInstaller.cs]: Shouldn't it be called "CopyKeyPropertyValues" or "PopulateKeyPropertyValues", as Get implies it should change its parameters

File: Composite/Data/DataFacade.cs [hunk 2]
```diff
@@ -509,7 +502,7 @@ public static LambdaExpression GetPredicateExpressionByUniqueKey(Type interfaceT
         // Private helper
         private static Expression GetPredicateExpressionByUniqueKeyFilterExpression(IReadOnlyList<PropertyInfo> keyProperties, DataKeyPropertyCollection dataKeyPropertyCollection, ParameterExpression parameterExpression)
         {
-            if (keyProperties.Count != dataKeyPropertyCollection.Count) throw new ArgumentException("Missing og to many key properties");
+            if (keyProperties.Count != dataKeyPropertyCollection.Count) throw new ArgumentException("Missing or to many key properties");
```
Comment (napernik) [inline Composite/Data/DataFacade.cs]: *too many

File: Composite/Data/DataFacade.cs [hunk 3]
```diff
@@ -606,6 +599,28 @@ public static IData TryGetDataByUniqueKey(Type interfaceType, DataKeyPropertyCol
             return data;
         }
 
+        /// <summary>
+        /// Returns all data items of the given type, which matchs the provided dataPropertyCollection (property/value pairs)
```
Comment (napernik) [inline Composite/Data/DataFacade.cs]: *matches

File: Composite/Core/PackageSystem/PackageFragmentInstallers/DataPackageFragmentInstaller.cs (hunk 1 above)
Comment (burningice2866) [reply Composite/Core/PackageSystem/PackageFragmentInstallers/DataPackageFragmentInstaller.cs]: Yeah, the method both returns the list of KeyProperties and populates them with values from the addElement-argument.

So maybe `PopulateAndReturnKeyPropertyValues`

**Pull request merged**